src/esn.calendar.libs/app/services/calendar-api.spec.js

```diff
@@ -643,6 +643,22 @@ describe('The calendar module apis', function() {
         this.$httpBackend.flush();
       });
     });
+
+    describe('the generate token for secret link request', function() {
+      it('should get the calendar export path', function() {
```

Comment:
You are not testing really testing the function here.

You need to write tests that make sure any changes to the code's logic will break the tests.

1. There is no testing for this part:

```js
.then(calHttpResponseHandler(200, _.property('data')))
```

2. There is no testing for this part:

```js
.catch(function(error) {
   notificationFactory.weakError('Failed to generate token for secret', 'Cannot join the server, please try later');

   return $q.reject(error);
});
```